Some new feature Backgroundless Pins, Journal Image by default, Journal Image Thumbnail and compatibility with FVTT 9

p4535992 opened the pull request:
Here I have tried to reduce the changes as much as possible.

I know for a fact that many people use the following three modules always together "pin-cushion", "foundryvtt-journal-thumbnail", "FVTT-Backgroundless-Pins" the last two have not been updated for almost 10 months and it made sense to me (also to avoid a module conflict) update and integrate these features.

I've checked the licenses and it should be fine.

- Add updated version of the module FVTT-Backgroundless-Pins integrated for avoid conflict (and because the module is not touched for 10 months)
- Add the setting to automatically updated by default the note image with the journal image if present
- Add updated version of the module foundryvtt-journal-thumbnail integrated for avoid conflict (and because the module is not touched for 10 months)

The combination of these features is in great demand in investigative games where you want to put objects in the room.

P.S. I add in the PR a prettier configuration based on what I have seen other foundry developers use, consider using it (or if you can please put your own on git) so as to synchronize us when we develop something and prevent git from going crazy with whitespace, tabs, etc.

p4535992 committed:
Add Backgrounless-Pins feature + update lang file the right way

p4535992 committed:
Add feature show journal thumbnail and add feature set by default the map note with the image of the journal

Commit:
[BETA] Add prettier plugin e merge icon player configuration from the pull request https://github.com/death-save/pin-cushion/pull/65

p4535992 committed:
Update and tested

p4535992 committed:
Bug fix

Commit:
Update README.md

p4535992 committed:
Some update

p4535992 committed:
Update

Commit:
Little bug fix

Commit:
Removed dist

Padhiver committed:
French translation

Hey ! 
I just translated for myself, so wanted to share with you if you want to. :)

Padhiver committed:
Updated for French translation.

Commit:
uPDATE

Commit:
Update

p4535992 committed:
Merge pull request #1 from Padhiver/patch-1

French translation

p4535992 committed:
Merge branch 'patch-2' of https://github.com/Padhiver/pin-cushion into Padhiver-patch-2

# Conflicts:
#	src/module.json

p4535992 committed:
Add french language

eclarke12 commented:
Thanks, I have done a quick review and it looks pretty good! I will pull and make some changes and aim to release an update later this week.

Ethaks commented on .gitignore:
```diff
@@ -116,4 +115,7 @@ dist
\ No newline at end of file
+foundry.js
+/.vscode
+/package-lock.json
```
Although this module is not particularly dependent on npm packages, is there a special reason to ignore the lockfile? It's usually recommended to commit it, and npm can use it for version checking even without `npm ci` (and not clutter commits with irrelevant changes).

p4535992 replied:
didn't know that, no problem to add the package-lock.json

Ethaks commented on package.json:
```diff
@@ -1,10 +1,12 @@
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "gulp build",
```
Possibly also a question @eclarke12 but does the gulp build process do something actually useful right now? I spy references to combat-carousel in there.

p4535992 replied:
i made a little fix for that

Ethaks commented on package.json:
```diff
@@ -25,6 +27,12 @@
     "gulp-zip": "^5.0.2",
     "jsdoc-to-markdown": "^6.0.1",
-    "node-fetch": "^2.6.1"
+    "node-fetch": "^2.6.1",
```
Just as a quick check: How many of these are used individually? Prettier is clear, but I'm unsure about the rest (and their need to be direct dependencies)

Ethaks commented on pin-cushion.js:
```diff
@@ -338,27 +456,43 @@ class PinCushion {
-            name: "SETTINGS.AboutAppN",
-            label: "SETTINGS.AboutAppN",
-            hint: "SETTINGS.AboutAppH",
+            name: game.i18n.localize("PinCushion.SETTINGS.AboutAppN"),
```
Is localizing these strings necessary? I vaguely remember Foundry doing that already on its own in the settings templates.

p4535992 replied:
not necessary , foundry should be try to localize the label anyway

Ethaks commented on pin-cushion.js:
```diff
@@ -504,6 +675,27 @@ class PinCushionHUD extends BasePlaceableHUD {
     }
 }
 
+class BackgroundlessControlIcon extends ControlIcon {
```
I guess one could argue about whether extending the class or wrapping the single changed method is the better thing to do here. I'm not sure about what other modules live in this space though, so I cannot really comment on the necessity to play nice with them.

p4535992 committed:
little update

Commit:
[BETA] Integration of poi-teleport

eclarke12 commented:
@p4535992 I haven't forgotten about this. I'm just finishing up a big CUB update then I will look at merging this PR.

p4535992 commented:
No problem man  CUB is much more important. 
By the way I committed a beta without thinking about it, tonight I'll fix it.

p4535992 committed:
add custom feature set image on journal when create a scene

p4535992 committed:
add image

p4535992 commented:
K the last commit is a starter integration of the "Point of Interest Teleport" module, for which I wanted to create a series of new features to integrate with pin cushion, so you might as well ignore it if you want. For now I will not update this PR again. Let me know if I need to fix anything for you. Greetings

Shuggaloaf commented:
Hi all! I actually came to death-save's git to request that he take up development of backgroundless pins because of it being abandoned. (And the pin cushion dev seemed like the logical place to start)

I saw this PR and couldn't believe it! Excellent job @p4535992 thank you so much for not just having the idea but helping make it happen. And you are right, I do also use journal thumbnails as well. Having all 3 in 1 mod, with reduced conflicts and mods count will be very nice. 

Thanks @eclarke12 for grabbing this and incorporating it. I just grabbed the updated CUB and so far it is great! (Thanks for fixing that double concentration issue). I'll be monitoring this for the merge. Looking forward to it.

Shuggaloaf commented:
@p4535992 / @eclarke12 May possibly want to consider not including the POI Teleport at this time. I say this hesitantly as I am 100% for combining as many mods as possible to reduce possible conflicts.

There are several other existing methods/mods to accomplish this. 
Trigger Happy mod can do this very easily just by adding a transparent drawing over the map pin area. When clicked (or optionally when a token is moved to that area), it loads whatever scene you would like it to. It also is not Scene Notes dependent.
Also, quite simply with no mods, you can just drag a scene into the journal entry of that map pin and create a link. Click to open map pin journal entry, click scene link, done. Usage is pretty much the same amount of clicks with no mod dependencies and less setup. 

Just some thoughts!

p4535992 committed:
update

p4535992 committed:
update

p4535992 committed:
update

p4535992 committed:
update

p4535992 committed:
version 1.5.3

p4535992 committed:
update README.md

p4535992 committed:
start compatibility foundryvtt 9

p4535992 committed:
update

p4535992 committed:
version 1.5.5

p4535992 committed:
version 1.5.6 beta

p4535992 committed:
bug fix on stupid flag

p4535992 committed:
version 1.5.6

p4535992 committed:
version 1.5.7

p4535992 committed:
update

p4535992 committed:
version 1.5.8

Ethaks commented:
@p4535992, would you consider the PR stable as is, or do you have plans for further additions etc.?
If you consider it to be somewhat done, I'll take another look and possibly change up some stuff/build upon it, before hopefully getting a release ready soon-ish.

p4535992 commented:
@Ethaks my PR was born to integrate some deprecated modules that I always used together (even if now they have updated them for 9) and synchronize everything with the wrapper module. Except for issues that are pointed out to me, I don't think I'll get back on it, for me you can try to take a look at it.

p4535992 committed:
version 1.5.9

p4535992 committed:
update CHANGELOG.md

p4535992 committed:
update CHANGELOG.md

PenguinMancer committed:
Foundry 9 Update Compatibility

Testing the latest version posted on Foundry 9 - Build 245, I noticed a few bugs.

Notably, trying to place more than one journal entry in a scene would result in the second marker not getting placed, and you not being able to place any more on the canvas until a page reload.

The cause of this seems to be how multiple times 'this.document.object.data' was used, and this was breaking things for reasons I don't understand. Changing these to 'this.document.getflag' seems to have fixed that issue entirely.

The await that was present at line 1204 was causing an error when placing a new marker stating that you must provide an _id for every new item in the update array. Commenting out that await cause the error to go away, and the marker will still update with the correct icon when hitting 'Update Map Note' to confirm placement anyway, so I don't see this as a huge loss.

The getFlag that was previously on line 1030 was returning as 'not a function', for whatever reason, but this is easily worked around by directly referencing the data in question.

Line 1039 has a simple change, from entities to documents. It stops a minor annoying warning in the console.

Finally, I checked compatibility with this for Foundry 0.8.9 and unfortunately it falls short. No horrible errors, but the 'show background?' checkbox in each marker does nothing at all, and the background is stuck as on. Because of that, I labeled this as a Foundry 9 exclusive update.

p4535992 committed:
Merge pull request #4 from PenguinMancer/master

Foundry 9 Update Compatibility

p4535992 committed:
merge pr to test

p4535992 committed:
version 1.5.10

p4535992 committed:
version 1.5.11

p4535992 committed:
clean up

p4535992 committed:
version 1.5.12

p4535992 committed:
a

p4535992 committed:
Merge branch 'master' of https://github.com/p4535992/pin-cushion

p4535992 commented:
@eclarke12 @Ethaks just a reminder for this, i made some test and seem working fine

p4535992 committed:
version 1.5.13

p4535992 committed:
version 1.5.14

p4535992 committed:
version 1.5.14

p4535992 committed:
bug fix to test

p4535992 committed:
version 1.5.15

p4535992 committed:
version 1.5.16

p4535992 committed:
integration new feature

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
u

p4535992 committed:
prettier

p4535992 committed:
beta ready

p4535992 committed:
beta ready 2

p4535992 committed:
version 1.6.0

p4535992 committed:
update README.md

p4535992 committed:
update README.md

p4535992 committed:
beta 1.6.1

p4535992 committed:
u

p4535992 committed:
version 1.6.1

p4535992 committed:
beta 1.6.2

p4535992 committed:
downgrade to 1.6.1

p4535992 committed:
version 1.6.2

p4535992 committed:
version 1.6.3

p4535992 committed:
some fix on height

p4535992 committed:
version 1.6.4

p4535992 committed:
u

p4535992 committed:
update

p4535992 committed:
prettier

p4535992 committed:
new feature

p4535992 committed:
ready to release

p4535992 committed:
adjust html code

p4535992 committed:
bug fix

p4535992 committed:
version 1.6.5

p4535992 committed:
add followmouse feature but is not working

p4535992 committed:
version 1.6.6

p4535992 committed:
bug fix

p4535992 committed:
version 1.6.7

p4535992 committed:
update module.json and module-dev.json

p4535992 committed:
version 1.6.8

p4535992 committed:
update README.md

p4535992 committed:
update README.md

p4535992 committed:
version 1.6.9

p4535992 committed:
version 1.6.10

p4535992 committed:
1.6.11

p4535992 committed:
version 1.6.12

p4535992 committed:
beta

p4535992 commented:
Now i'm officially the new maintainer of the module so i'll close this PR

p4535992 committed:
beta

p4535992 committed:
prettier

p4535992 committed:
version 1.7.0

p4535992 commented:
I'll close this pr because, i'm now the officialy new maintainer.
